src/ort_genai_c.h

```diff
@@ -200,6 +200,13 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateConfig(const char* config_path, OgaC
  */
 OGA_EXPORT OgaResult* OGA_API_CALL OgaConfigClearProviders(OgaConfig* config);
 
+/**
+ * \brief Get the Model Type in the given config
+ * \param[in] config The config to get the model type from
+ * \return OgaResult containing the error message if the clearing of the providers failed.
+ */
+OGA_EXPORT OgaResult* OGA_API_CALL OgaConfigGetModelType(OgaConfig* config, std::string& name);
```

Comment:
std::string cannot be used in C since it is a C++ concept. We would need to use char** here instead.